src/components/Budget/BudgetView/BudgetViewContainer.js

```diff
@@ -44,17 +46,20 @@ const BudgetViewContainer = ({ history, location, match, mutator }) => {
   const [budget, setBudget] = useState({});
   const [fiscalYear, setFiscalYear] = useState();
   const [isLoading, setIsLoading] = useState(true);
+  const [expenseClassesTotals, setExpenseClassesTotals] = useState();
   const showCallout = useShowCallout();
 
   const fetchBudgetResources = useCallback(
     () => {
       setIsLoading(true);
       setBudget({});
+      setExpenseClassesTotals();
       setFiscalYear({});
 
-      mutator.budgetById.GET()
-        .then(budgetResponse => {
+      Promise.all([mutator.budgetById.GET(), mutator.expenseClassesTotals.GET()])
```

Comment:
Could you please add catch that something was not loaded? Because for user it will looks like all is fine.